Fix compiler warning in Mapper::open_

The recent changes have the following compiler warning. I recommend commenting out the "flags" parameter, using it, or removing parameter.

```
lib/marisa/grimoire/io/mapper.cc:134:46: warning: unused parameter 'flags' [-Wunused-parameter]
  134 | void Mapper::open_(const char *filename, int flags) {
      |                                              ^
1 warning generated.
```

--- a/lib/marisa/grimoire/io/mapper.cc
+++ b/lib/marisa/grimoire/io/mapper.cc
@@ -126,6 +126,13 @@ void Mapper::open_(const char *filename, int flags) {
   origin_ = ::MapViewOfFile(map_, FILE_MAP_READ, 0, 0, 0);
   MARISA_THROW_IF(origin_ == NULL, MARISA_IO_ERROR);
 
+  if (flags & MARISA_MAP_POPULATE) {
+    WIN32_MEMORY_RANGE_ENTRY range_entry;
+    range_entry.VirtualAddress = origin_;
+    range_entry.NumberOfBytes = size_;
+    PrefetchVirtualMemory(GetCurrentProcess(), 1, &range_entry, 0);
+  }
+
   ptr_ = static_cast<const char *>(origin_);
   avail_ = size_;
 }

--- a/lib/marisa/grimoire/io/mapper.cc
+++ b/lib/marisa/grimoire/io/mapper.cc
@@ -140,10 +140,15 @@ void Mapper::open_(const char *filename, int flags) {
   size_ = (std::size_t)st.st_size;
 
   int map_flags = MAP_SHARED;
+ #if defined(MAP_POPULATE)
   // `MAP_POPULATE` is Linux-specific.
- #ifdef MAP_POPULATE
   if (flags & MARISA_MAP_POPULATE) {
     map_flags |= MAP_POPULATE;
+  }
+ #elif defined(MAP_PREFAULT_READ)
+  // `MAP_PREFAULT_READ` is FreeBSD-specific.
+  if (flags & MARISA_MAP_POPULATE) {
+    map_flags |= MAP_PREFAULT_READ;
   }
  #endif
 

